lib/helpers/ptcpnt_persn_id_depnt_org_fix.rb

```diff
@@ -0,0 +1,170 @@
+# frozen_string_literal: true
+
+class PtcpntPersnIdDepntOrgFix < CaseflowJob
+  ERROR_TEXT = "participantPersonId does not match a dependent or an organization"
+
+  ASSOCIATIONS = [
+    BgsPowerOfAttorney,
+    BgsAttorney,
+    CavcRemandsAppellantSubstitution,
+    Claimant,
+    DecisionIssue,
+    EndProductEstablishment,
+    Notification,
+    Organization,
+    Person,
+    RequestIssue,
+    VbmsDistribution,
+    Veteran
+  ].freeze
+
+  def initialize(stuck_job_report_service)
+    @stuck_job_report_service = stuck_job_report_service
+  end
+
+  def start_processing_records
+    return if self.class.error_records.blank?
+
+    # count of records with errors before fix
+    @stuck_job_report_service.append_record_count(self.class.error_records.count, ERROR_TEXT)
+
+    self.class.error_records.each do |supp_claim|
+      incorrect_pid = supp_claim.claimant.participant_id
+      # check that claimant type is VeteranClaimant
+      next unless supp_claim.claimant.type == "VeteranClaimant"
+
+      veteran_file_number = supp_claim.veteran.file_number
+      @correct_pid = retrieve_correct_pid(veteran_file_number)
+
+      handle_person_and_claimant_records(supp_claim)
+      retrieve_records_to_fix(incorrect_pid)
+
+      @stuck_job_report_service.append_single_record(supp_claim.class.name, supp_claim.id)
+      # Re-run job after fixing broken records
+      re_run_job(supp_claim)
+    end
+    # record count with errors after fix
+    @stuck_job_report_service.append_record_count(self.class.error_records.count, ERROR_TEXT)
+  end
+
+  class << self
+    # def get_correct_person(correct_pid)
+    #   Person.find_by(participant_id: correct_pid)
+    # end
+
+    def iterate_through_associations_with_bad_pid(incorrect_pid, incorrectly_associated_records)
+      ASSOCIATIONS.each do |ass|
+        if ass.attribute_names.include?("participant_id")
+          records = ass.where(participant_id: incorrect_pid)
+          incorrectly_associated_records.push(*records)
+
+        elsif ass.attribute_names.include?("claimant_participant_id")
+          records = ass.where(claimant_participant_id: incorrect_pid)
+          incorrectly_associated_records.push(*records)
+        elsif ass.attribute_names.include?("veteran_participant_id")
+          records = ass.where(veteran_participant_id: incorrect_pid)
+          incorrectly_associated_records.push(*records)
+        end
+      end
+    end
+
+    def error_records
+      SupplementalClaim.where("establishment_error ILIKE ?", "%#{ERROR_TEXT}%")
```

Comment:
do you want to exclude canceled establishments?
```suggestion
      SupplementalClaim.where("establishment_error ILIKE ?", "%#{ERROR_TEXT}%").where(establishment_canceled_at: nil)
```